docs/framework/react/guides/query-invalidation.md

```diff
@@ -131,3 +134,20 @@ const todoListQuery = useQuery({
 ```
 
 [//]: # 'Example5'
+
+By default, `invalidateQueries` only refetches active queries. If you want to refetch **all matching queries**, including those currently inactive in the cache, you need to explicitly set the `refetchType` (or `type`) option to `'all'`:
+
+[//]: # 'Example6'
+
+```tsx
+queryClient.invalidateQueries({
+  queryKey: ['todos'],
+  refetchType: 'all' // or type: 'all'
+});
+```
+
+[//]: # 'Example6'
+
+This forces every matching query—active or inactive—to be refetched immediately.
+
+> 🧠 **Note:** The default behavior (`refetchType: 'active'`) helps avoid unnecessary network requests. But if you're triggering an invalidation after a mutation that affects *all clients or views*, make sure to use `refetchType: 'all'` to keep all query caches up to date.
```

Comment:
I don’t understand that sentence - what is “all clients or views” referring to? Imo there’s really no need to set `refetchType: 'all'`, as it will likely lead to overfetching. It's something we have documented in the [API docs](https://tanstack.com/query/v5/docs/reference/QueryClient#queryclientinvalidatequeries), but I don’t think we need to emphasize on this with a guide.